core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -374,4 +374,8 @@ TableScanContext reportWith(MetricsReporter reporter) {
         fromSnapshotInclusive,
         reporter);
   }
+
+  public void setColStats(boolean colStats) {
```

Comment:
This class is not mutable. Can you update this to follow the same convention as other methods that return a new `TableScanContext` with the option set?

Reply:
Fixed